lib/KBox/src/KMessage.h

```diff
@@ -191,6 +196,94 @@ class IMUMessage: public KMessage {
     };
 };
 
+class NAVMessage: public KMessage {
+  private:
+    bool apMode, apHeadingMode, apWaypointMode, apDodgeMode;
+    double currentHeading, targetHeading, courseToWaypoint;
```

Comment:
I did some research on how autopilot are defined by other protocols.

- In [NMEA0183](http://catb.org/gpsd/NMEA.html#_apb_autopilot_sentence_b) the key sentence is APB which is sent by a GPS or a control unit and tells the pilot which heading to follow 
- Or [RMB](http://catb.org/gpsd/NMEA.html#_apb_autopilot_sentence_b) which is very similar. [Apparently that is the one OpenCPN uses](.
- [SignalK defines 8 possible "states"](https://github.com/SignalK/specification/blob/master/schemas/groups/steering.json#L33): "auto", "standby", "alarm", "noDrift", "wind", "depthContour", "route", "directControl"
- [SignalK defines 3 possible "target"](https://github.com/SignalK/specification/blob/master/schemas/groups/steering.json#L73) which are headingMagnetic, headingTrue, windAngleApparent
 - [This SignalK plugin](https://github.com/sbender9/signalk-raymarine-autopilot/blob/master/index.js) has NMEA2000 commands to control a Raymarine autopilot. The four supported states are auto/route/wind/standby.

There are a few things I draw from this research:
 - Most current autopilots do not do navigation calculation to a waypoint. They rely on an external device (GPS unit, PC, iPad, etc) to tell them what is the heading to destination and they follow it.
 - It would be really cool to display this information on KBox so that whether you have a human pilot or an autopilot, you could see on KBox how well you are doing towards your waypoint, whether you are on course and how far you are.
 - The way to "dodge" with a traditional autopilot is to press -10/+10 (see [this discussion](http://www.cruisersforum.com/forums/f134/avoid-function-with-opencpn-and-autopilot-140503.html)). This could be done by just turning the rotary button on KBox.
 - We could have an extra "follow waypoint" feature where KBox would be smart enough to be both the autopilot control head and the navigation system. We would have to include also detecting when you get in the arrival circle or the perpendicular of the waypoint which I think they do by keeping the original heading to target. If we have this feature, dodge makes a lot of sense.

So coming back to our current pull request :) All this research helped me realize that what we are doing here is "just" the autopilot head control. Not the navigation part. I agree we want to do the navigation part at some point, but to do that we will need to be able to parse incoming messages to get current position and position of a target waypoint. Let's build things one step at a time.

So this is what I suggest and will push in a different branch for you to review:
 - Rename NavigationPage to AutopilotControlPage, it will be dedicated to controlling an Autopilot from KBox.
 - Remove reference to waypoint for now, focus on having a targetMagneticHeading and building an autopilot that can follow it.